Modules/Loadable/Markups/VTKWidgets/vtkSlicerAngleWidget.cxx

```diff
@@ -38,6 +38,22 @@ vtkSlicerAngleWidget::vtkSlicerAngleWidget()
 //----------------------------------------------------------------------
 vtkSlicerAngleWidget::~vtkSlicerAngleWidget() = default;
 
+//----------------------------------------------------------------------
+vtkSlicerMarkupsWidget* vtkSlicerAngleWidget::CreateInstance() const
```

Comment:
I'm just curious, have you tried using vtkObjectFactory for this?
Since this code needs to be repeated in each class, it would be better to create a macro and/or template function so that it can be implemented by a single line of code (probably in the header). It would also allow us to change the instantiation at a single place, if a change is needed in the future.

Reply:
@lassoan I did try with `vtkObjectFactory` but it seems it does not operate on `vtkSlicerWidget`s. I have created a macro that is used in the widget's header file (single line).

Reply:
I tried `vtkObjectFactory`, but it does not seem to operate in Markups VTKWidgets. I created a macro to have a 1-line declaration in the VTKWidgets header files as you suggest.